sympy/integrals/transforms.py

```diff
@@ -147,18 +166,13 @@ def doit(self, **hints):
         ``(simplify, noconds, needeval) = (True, False, False)``.
         """
         needeval = hints.pop('needeval', False)
-        try_directly = not any(func.has(self.function_variable)
-                               for func in self.function.atoms(AppliedUndef))
-        if try_directly:
-            try:
-                return self._compute_transform(self.function,
-                    self.function_variable, self.transform_variable, **hints)
-            except IntegralTransformError:
-                pass
+        simplify = hints.pop('simplify', True)
+        hints['simplify'] = simplify
 
-        fn = self.function
-        if not fn.is_Add:
-            fn = expand_mul(fn)
+        fn, T = self._try_directly(**hints)
+
+        if not T==None:
+            return T
```

Comment:
Not so important but `if T is not None` is usually the preferred way to spell this.

Reply:
I will change it everywhere, it looks nicer.